integration/solana/setup.ts

```diff
@@ -1,46 +1,78 @@
 // SPDX-License-Identifier: Apache-2.0
 
-import { Connection, Keypair, LAMPORTS_PER_SOL, BpfLoader, BPF_LOADER_PROGRAM_ID } from '@solana/web3.js';
-import { Contract } from '@solana/solidity';
+import { Connection, Keypair, LAMPORTS_PER_SOL, PublicKey, BpfLoader, Transaction, SystemProgram, BPF_LOADER_PROGRAM_ID } from '@solana/web3.js';
+import { AnchorProvider, Program } from '@project-serum/anchor';
 import fs from 'fs';
 
 const endpoint: string = process.env.RPC_URL || "http://127.0.0.1:8899";
 
 export async function loadContract(name: string, args: any[] = [], space: number = 8192):
-    Promise<{ contract: Contract, connection: Connection, payer: Keypair, program: Keypair, storage: Keypair }> {
+    Promise<{ program: Program, payer: Keypair, provider: AnchorProvider, storage: Keypair, program_key: PublicKey }> {
 
-    const abi = JSON.parse(fs.readFileSync(`${name}.abi`, 'utf8'));
+    const idl = JSON.parse(fs.readFileSync(`${name}.json`, 'utf8'));
 
-    const connection = newConnection();
+    const payer = loadKey('payer.key');
 
-    const payerAccount = loadKey('payer.key');
-    const program = loadKey(`${name}.key`);
+    process.env['ANCHOR_PROVIDER_URL'] = endpoint;
+    process.env['ANCHOR_WALLET'] = 'payer.key';
+
+    const provider = AnchorProvider.env();
```

Comment:
`const provider = new AnchorProvider(arguments_here)` requires a connection. I've used `.local(endpoint)` so that the `ANCHOR_PROVIDER_URL` is not needed.